addons/mail_group/views/mail_group_views.xml

```diff
@@ -6,19 +6,54 @@
         <field name="arch" type="xml">
             <tree>
                 <field name="name"/>
+                <field name="alias_name" string="Alias"/>
+                <field name="moderation" string="Moderated"/>
+                <field name="member_count" string="Members"/>
             </tree>
         </field>
     </record>
+    <record id="mail_group_view_kanban" model="ir.ui.view">
+        <field name="name">mail.group.view.kanban</field>
+        <field name="model">mail.group</field>
+        <field name="arch" type="xml">
+            <kanban string="Mail Groups">
+                <field name="id"/>
+                <field name="description"/>
+                <field name="is_member"/>
+                <templates>
+                    <t t-name="kanban-description">
+                        <div class="oe_group_description" t-if="record.description.raw_value">
+                            <field name="description"/>
+                        </div>
+                    </t>
+                    <t t-name="kanban-box">
+                        <div class="oe_module_vignette oe_kanban_global_click">
+                            <img t-att-src="kanban_image('mail.group', 'image_128', record.id.raw_value)" class="oe_module_icon" alt="Group"/>
+                            <div class="oe_module_desc">
+                                <h4 class="o_kanban_record_title">#<field name="name"/></h4>
+                                <p class="o_mg_description text-truncate text-nowrap" t-esc="record.description.raw_value or ''"/>
+                                <field name="is_member" invisible="1"/>
+                                <button type="object" attrs="{'invisible':[('is_member', '=', True)]}" class="btn btn-primary float-right" name="action_join">Join</button>
+                                <button type="object" attrs="{'invisible':[('is_member', '=', False)]}" class="btn btn-secondary float-right" name="action_leave">Leave</button>
+                            </div>
+                        </div>
+                    </t>
+                </templates>
+            </kanban>
+        </field>
+    </record>
     <record id="mail_group_view_form" model="ir.ui.view">
         <field name="name">mail.group.view.form</field>
         <field name="model">mail.group</field>
         <field name="arch" type="xml">
             <form string="Mail Group">
                 <header>
-                    <button name="action_send_guidelines" type="object" class="btn btn-secondary" string="Send Guidelines" attrs="{'invisible': ['|', ('is_moderator', '=', False), ('moderation', '=', False)]}"/>
+                    <button type="object" attrs="{'invisible':[('is_member', '=', True)]}" class="btn btn-primary" name="action_join" string="Join"/>
```

Comment:
Linting: several buttons / new tags could be multi lined (going out of my screen when reviewing generally indicates we could split lines :D )